relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -190,96 +263,75 @@ impl TraceContext {
     /// Returns the decision of whether to sample or not a trace based on the configuration rules
     /// If None then a decision can't be made either because of an invalid of missing trace context or
     /// because no applicable sampling rule could be found.
-    fn should_sample(
-        &self,
-        config: &SamplingConfig,
-        project_id: ProjectId,
-        strategy: SamplingStrategy,
-    ) -> Option<bool> {
-        let rule = get_matching_rule(config, self, project_id, strategy)?;
+    fn should_sample(&self, config: &SamplingConfig) -> Option<bool> {
+        let rule = get_matching_rule(config, self, RuleType::Trace)?;
         let rate = pseudo_random_from_uuid(self.trace_id)?;
         Some(rate < rule.sample_rate)
     }
 }
 
-impl SamplingContextProvider for TraceContext {
-    fn release(&self) -> Option<&str> {
-        self.release.as_deref()
-    }
-
-    fn environment(&self) -> Option<&str> {
-        self.environment.as_deref()
-    }
-
-    fn user_segment(&self) -> Option<&str> {
-        self.user_segment.as_deref()
-    }
-}
-
-/// NOTE: since relay-general doesn't know anything about dynamic sampling
-/// SamplingContextProvider for Event is implemented here.
-impl SamplingContextProvider for Event {
-    fn release(&self) -> Option<&str> {
-        self.release.as_str()
-    }
-
-    fn environment(&self) -> Option<&str> {
-        self.environment.as_str()
-    }
-
-    fn user_segment(&self) -> Option<&str> {
-        None // TODO RaduW at the moment (10.12.2020) we don't have this (discussions pending)
-    }
-}
-
 // Checks whether an event should be kept or removed by dynamic sampling
 pub fn should_keep_event(
     event: &Event,
     project_state: &ProjectState,
-    project_id: ProjectId,
+    processing_enabled: bool,
```

Comment:
Instead of passing `processing_enabled`, should we name this `skip_unsupported`?

Reply:
I think that this makes it clear the business rule (that when processing is not enabled and we have unsupported rules we do not sample). If we call the parameter skip_unsupported and we pass in it processing_enabled flag we need to navigate to the call site to infer the business rule (i.e. it is not clear anymore when we read the function why we are doing it).

If you feel strongly about it I'll change it to `processing_enabled`.

Reply:
I do not feel strongly about this, and your argument makes sense. Let's keep as is.